tools/loader/ImageClassifier.cpp

```diff
@@ -291,25 +333,41 @@ int main(int argc, char **argv) {
   // if requested from command line.
   loader.compile(ctx);
 
+  assert(!(streamInputFilenamesMode && emittingBundle()) &&
+         "Cannot emit a bundle and also stream inputs.");
+
   // If in bundle mode, do not run inference.
   if (!emittingBundle()) {
+    while (true) {
+      // Update the inputs.
+      updateInputPlaceholders(ctx, {inputImage}, {&data});
+
+      // Perform the inference execution.
+      loader.runInference(ctx);
+
+      // Print out the inferred image classification.
+      auto H = SMVarT->getHandle<>();
+      llvm::outs() << "Model: " << loader.getFunction()->getName() << "\n";
+      for (unsigned i = 0; i < inputImageFilenames.size(); i++) {
+        Tensor slice = H.extractSlice(i);
+        auto SH = slice.getHandle<>();
+        llvm::outs() << " File: " << inputImageFilenames[i];
+
+        auto topKPairs = getTopKPairs(SH);
+        printTopKPairs(topKPairs);
+      }
 
-    // Update the inputs.
-    updateInputPlaceholders(ctx, {inputImage}, {&data});
-
-    // Perform the inference execution.
-    loader.runInference(ctx);
+      // If not in streaming mode then we just run inference once and exit.
+      if (!streamInputFilenamesMode) {
+        break;
+      }
 
-    // Print out the inferred image classification.
-    auto H = SMVarT->getHandle<>();
-    llvm::outs() << "Model: " << loader.getFunction()->getName() << "\n";
-    for (unsigned i = 0; i < inputImageFilenames.size(); i++) {
-      Tensor slice = H.extractSlice(i);
-      auto SH = slice.getHandle<>();
-      llvm::outs() << " File: " << inputImageFilenames[i];
+      // Load the next round of images to infer.
+      getNextImageFilenames(&inputImageFilenames);
+      loadImagesAndPreprocess(inputImageFilenames, &data);
```

Comment:
Would it be possible to do that at the beginning of the loop so that we don't have to repeat it outside of the loop?

Reply:
Because we need to know what our input shape is before compilation, I think it needs to be done once prior to the loop.

The other option would be to add the input shape as a configurable command line parameter, then this could be done once, at the start of this loop. I didn't think that was a great path forward. Though we could set `{c,h,w}` to the imagenet defaults `{3, 224, 224}` and so batch would be the only one changed unless running on mnist or another dataset.

Reply:
Alternatively we could move the logic doing the compilation inside the loop and guard it with a `isFirst` kind of thing.